File: binarysearch.cpp
#include<iostream>
#include<conio.h>
using namespace std;
int binarysearch(int *a,int l,int r,int x)
{
    int mid=(l+r)/2;
    if(x<a[mid])
        return binarysearch(a,l,mid-1,x);
    if(x>a[mid])
        return binarysearch(a,mid+1,r,x);
    if(x==a[mid])
        return mid;
    return -1;
}
int main()
{
    int ans,n,x;
    cout<<"Enter the size of the array";
    cin>>n;
    cout<<"Enter the values in the array";
    int t,a[n];
    for(i=0;i<n;i++)
    {
        cin>>a[i];
    }
    cout<<"Enter the element you want to find"<<endl;
    cin>>x;
    ans=binarysearch(a,0,n-1,x);
    if(ans==-1)
        cout<<"Element not found"<<endl;
    else
        cout<<"Element found at "<<ans<<"position;
    return 0;
}

File: Searching Algo/BinarySearch.py
def binary_search (arr, l, r, x): 
    if r >= l: 
        mid = int(l + (r - l)/2)  
        if arr[mid] == x: 
            return mid 
        elif arr[mid] > x: 
            return binary_search(arr, l, mid-1, x) 
        else: 
            return binary_search(arr, mid + 1, r, x)   
    else: 
        return -1
 
arr = [ 2, 3, 4, 10, 40 ] 
x = 10
result = binary_search(arr, 0, len(arr)-1, x) 
  
if result != -1: 
    print("Element is present at index",result) 
else: 
    print("Element is not present in array")